### PR TITLE
Removed user and group details from logs

### PR DESCRIPTION
**Pull request by shtripat**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by klauspost** [comment]
Aren't most of these coming from the request itself? I don't see any problem in returning those cases. The error message is after all clearer when it is included.

File: cmd/admin-handlers-idp-ldap.go [hunk 1]
```diff
@@ -292,7 +292,7 @@ func (a adminAPIHandlers) AddServiceAccountLDAP(w http.ResponseWriter, r *http.R
 			return
 		}
 		if len(ldapPolicies) == 0 {
-			err = fmt.Errorf("No policy set for user `%s` or any of their groups: `%s`", opts.claims[ldapActualUser], strings.Join(targetGroups, "`,`"))
+			err = fmt.Errorf("No policy set for user or any of their groups")
```
Comment (klauspost) [inline cmd/admin-handlers-idp-ldap.go]: Comes from req AFAICT

File: cmd/admin-handlers-users.go [hunk 2]
```diff
@@ -669,7 +669,7 @@ func (a adminAPIHandlers) AddServiceAccount(w http.ResponseWriter, r *http.Reque
 		_, isRegularUser := globalIAMSys.GetUser(ctx, targetUser)
 		if !isRegularUser && targetUser != globalActiveCred.AccessKey {
 			apiErr := toAdminAPIErr(ctx, errNoSuchUser)
-			apiErr.Description = fmt.Sprintf("Specified target user %s does not exist", targetUser)
+			apiErr.Description = fmt.Sprintf("Specified target user does not exist")
 			writeErrorResponseJSON(ctx, w, apiErr, r.URL)
```
Comment (klauspost) [inline cmd/admin-handlers-users.go]: Provided by req. We can return it.

File: cmd/admin-handlers-users.go [hunk 3]
```diff
@@ -2308,7 +2308,7 @@ func (a adminAPIHandlers) ImportIAM(w http.ResponseWriter, r *http.Request) {
 					// clean import.
 					err := globalIAMSys.DeleteServiceAccount(ctx, svcAcctReq.AccessKey, true)
 					if err != nil {
-						delErr := fmt.Errorf("failed to delete existing service account(%s) before importing it: %w", svcAcctReq.AccessKey, err)
+						delErr := fmt.Errorf("failed to delete existing service account before importing it: %w", err)
```
Comment (klauspost) [inline cmd/admin-handlers-users.go]: Comes from req.

File: cmd/sts-handlers.go [hunk 4]
```diff
@@ -672,8 +672,7 @@ func (sts *stsAPIHandlers) AssumeRoleWithLDAPIdentity(w http.ResponseWriter, r *
 	}
 	if len(ldapPolicies) == 0 && newGlobalAuthZPluginFn() == nil {
 		writeSTSErrorResponse(ctx, w, ErrSTSInvalidParameterValue,
-			fmt.Errorf("expecting a policy to be set for user `%s` or one of their groups: `%s` - rejecting this request",
-				ldapActualUserDN, strings.Join(groupDistNames, "`,`")))
+			fmt.Errorf("expecting a policy to be set for user or one of their groups - rejecting this request"))
```
Comment (klauspost) [inline cmd/sts-handlers.go]: Admin req, not logged.

**Review by klauspost — COMMENTED** [review]
Responses where the data comes from requst.

**Comment by shtripat** [comment]
> Aren't most of these coming from the request itself? I don't see any problem in returning those cases. The error message is after all clearer when it is included.

Thanks for guidance. Incorporated the suggestions.

**Review by klauspost — APPROVED** [review]
LGTM

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**